### PR TITLE
[SLP]Correctly detect minnum/maxnum patterns for select/cmp operations on floats.

### PR DESCRIPTION
**Pull request by alexey-bataev**
The patch enables detection of minnum/maxnum patterns for float point
instruction, represented as select/cmp. Also, enables better cost
estimation for integer min/max patterns since the compiler starts
to estimate the scalars separately.

### CODE REVIEW

**Comment by alexey-bataev** [comment]
Ping!

File: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp [hunk 1]
```diff
@@ -9636,9 +9636,27 @@ BoUpSLP::getEntryCost(const TreeEntry *E, ArrayRef<Value *> VectorizedVals,
                                        ? CmpInst::BAD_FCMP_PREDICATE
                                        : CmpInst::BAD_ICMP_PREDICATE;
 
-      return TTI->getCmpSelInstrCost(E->getOpcode(), OrigScalarTy,
-                                     Builder.getInt1Ty(), CurrentPred, CostKind,
-                                     VI);
+      InstructionCost ScalarCost = TTI->getCmpSelInstrCost(
+          E->getOpcode(), OrigScalarTy, Builder.getInt1Ty(), CurrentPred,
+          CostKind, VI);
+      auto IntrinsicAndUse = canConvertToMinOrMaxIntrinsic(VI);
```
Comment (RKSimon) [inline llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:9642]: Is this any cleaner if we use a binding? (and below in GetVectorCost).
```
auto [MinMaxID, SelectOnly] = canConvertToMinOrMaxIntrinsic(VI);
```

**Review by RKSimon — APPROVED** [review]
LGTM with one minor thought

File: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp (hunk 1 above)
Comment (alexey-bataev) [reply llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp]: Done

**Pull request merged**